src/drive/ducks/files/Toolbar.jsx

```diff
@@ -58,146 +97,97 @@ class Toolbar extends Component {
       canUpload,
       uploadFiles,
       downloadAll,
-      leaveFolder,
-      breakpoints: { isMobile }
+      leaveFolder
     } = this.props
     const notRootfolder = displayedFolder && displayedFolder.id !== ROOT_DIR_ID
     const hasWriteAccess =
-      canUpload && (!shared.withMe || shared.sharingType === 'master-master')
+      canUpload && (!shared.withMe || shared.sharingType === 'two-way')
 
-    const MoreMenu = (
-      <Menu
-        title={t('toolbar.item_more')}
-        disabled={disabled}
-        className={styles['fil-toolbar-menu']}
-        button={<MoreButton>{t('Toolbar.more')}</MoreButton>}
-      >
-        {notRootfolder &&
-          !shared.withMe && (
-            <Item>
-              <a
-                className={styles['fil-action-share']}
-                onClick={() => this.setState(toggleShowShareModal)}
-              >
-                {t('toolbar.share')}
-              </a>
-            </Item>
-          )}
-        {shared.withMe && (
-          <Item>
-            <a
-              className={styles['fil-action-share']}
-              onClick={() => this.setState(toggleShowShareModal)}
+    return (
+      <div className={styles['fil-toolbar-files']}>
+        <Toolbar
```

Comment:
So you want to create a Toolbar, which has one or more highlighted actions, and several more inside the "more" menu. On mobile, all actions are inside the "more" menu.
This is how you use the API:

- create a `Toolbar` container
- list your actions as `ToolbarAction`, indicating if they should be visible and/or highlighted
- provide a Component to render for the `ToolbarAction` as a child

For actions that are always in the "more" menu, it's very easy, you just do something like

```
<ToolbarAction>
  <a className onClick>{label}</a>
<ToolbarAction/>
```

This could be made even easier by adding props to the `ToolbarAction` like `icon`, `label`, etc.

Things are trickier for actions that are renderer both outside *and* inside the menu, since they might look quite different and the `ToolbarAction` has no idea. In those cases, the components receive a `highlighted` prop, which tells them if they are renderer inside or outside of the menu.